src/vs/workbench/contrib/terminal/browser/links/terminalLinkOpeners.ts

```diff
@@ -218,6 +218,15 @@ export class TerminalSearchLinkOpener implements ITerminalLinkOpener {
 
 	private async _tryOpenExactLink(text: string, link: ITerminalSimpleLink): Promise<boolean> {
 		const sanitizedLink = text.replace(/:\d+(:\d+)?$/, '');
+		// For only file links disallow exact link matching, for example searching for `foo.txt`
+		// when no cwd information is available should search when only the initial cwd is available
+		// as it's ambiguous if there are multiple matches.
+		//
```

Comment:
this is confusing to me

Reply:
so when no cwd is available, search using initialCwd?

Reply:
Yeah that was a bit rough lol, what do you think of the updated one?